### PR TITLE
feat(telemetry): Include user intent for copy and insert code interactions

### PR DESCRIPTION
**Pull request by lalexcaws**
## Problem
Need to gather acceptance metrics on generated unit tests

## Solution
Include user intent on copy and insert code interactions in telemetry to form an acceptance rate

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/shared/telemetry/vscodeTelemetry.json [hunk 1]
```diff
@@ -821,6 +821,10 @@
                 {
                     "type": "cwsprChatMessageId"
                 },
+                {
+                    "type": "cwsprChatUserIntent",
+                    "required": false
+                },
```
Comment (hayemaxi) [inline packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: Do you have a plan/backlog item to port this to aws-toolkit-common?

Comment (lalexcaws) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: No, I missed the upstreaming part. Should I create an issue for tracking this?

Comment (justinmk3) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: Yes, that is required for all metrics (with rare exceptions). All of the amazon q metrics in this file need to be upstreamed (and aligned with other IDEs), assuming they are still in use.

Comment (hayemaxi) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: Yes, after merging this you should treat it as high priority to move this metric to commons, especially if other IDEs need to use it.

Comment (lalexcaws) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: Got it, I will take that as an action item and work on it

**Review by laileni-aws — APPROVED** [review]

File: packages/core/src/shared/telemetry/vscodeTelemetry.json (hunk 1 above)
Comment (lalexcaws) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:827]: Update for aws-toolkit-common: https://github.com/aws/aws-toolkit-common/pull/864

**Pull request merged**